archive/tar_stream.ts

```diff
@@ -0,0 +1,339 @@
+// Copyright 2018-2024 the Deno authors. All rights reserved. MIT license.
+/**
+ * The interface required to provide a file.
+ */
+export interface TarStreamFile {
+  pathname: string | [Uint8Array, Uint8Array];
```

Comment:
Since validation of the pathnames happened inside the TarStream, it would throw an error if it was invalid. I exposed the `parsePathname()` function so a user could choose to validate the names before piping them through to avoid having the entire stream crash midway, and then also decided to allow the user to pass in that result to save on having to validate the pathname again inside the TarStream.